iter/iter.go

```diff
@@ -9,28 +9,64 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
+// defaultMaxGoroutines returns the default maximum number of
+// goroutines to use within this package.
+func defaultMaxGoroutines() int { return runtime.GOMAXPROCS(0) }
+
+// Iterator can be used to configure the behaviour of ForEach
+// and ForEachIdx. The zero value is safe to use with reasonable
+// defaults.
+//
+// Iterator is safe for reuse and concurrent use.
+type Iterator[T any] struct {
+	// MaxGoroutines controls the maximum number of goroutines
+	// to use on this Iterator's methods.
+	//
+	// If unset, MaxGoroutines defaults to runtime.GOMAXPROCS(0).
+	MaxGoroutines int
```

Comment:
I originally named this `Concurrency`, but renamed it to `MaxGoroutines` to align it with `(*Pool).WithMaxGoroutines()`